tmt/steps/prepare/artifact/providers/repository_provider.py

```diff
@@ -1,108 +1,139 @@
 """
-Artifact providers for Repository and Repository Files.
+Artifact provider for repository files.
 """
 
 from collections.abc import Iterator
+from re import Pattern
 from shlex import quote
-from urllib.parse import urlparse
+from typing import Any, Optional
+from urllib.parse import unquote, urlparse
 
-import tmt.log
-import tmt.utils
-from tmt.container import container
 from tmt.steps.prepare.artifact.providers import (
     ArtifactProvider,
     DownloadError,
 )
-from tmt.steps.prepare.artifact.providers.info import ArtifactInfo
+from tmt.steps.prepare.artifact.providers.info import RpmArtifactInfo
 from tmt.steps.provision import Guest
 from tmt.utils import GeneralError, Path, ShellScript
 
 
-@container
-class RepositoryFileInfo(ArtifactInfo):
+class RepositoryFileInfo:
```

Comment:
Do we need the helper class? Seems like most of it can be inlined in `RepositoryFileProvider`. Otherwise rename it to `RepositoryFile` in order to avoid confusion